src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenConverter.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.IdentityModel.JsonWebTokens;
+
+namespace System.IdentityModel.Tokens.Jwt
+{
+    /// <summary>
+    /// Static class to convert a <see cref="JsonWebToken"/> to a <see cref="JwtSecurityToken"/>
+    /// </summary>
+    public static class JwtSecurityTokenConverter
+    {
+        /// <summary>
+        /// Initializes a new instance of a <see cref="JwtSecurityToken"/> from a <see cref="JsonWebToken"/>
+        /// </summary>
+        /// <param name="token">A JSON Web Token to convert from.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="token"/> is null</exception>
+        public static JwtSecurityToken Convert(JsonWebToken token)
+        {
+            if (token == null)
+                throw new ArgumentNullException(nameof(token));
+
+            if (token.InnerToken != null)
+            {
+                var outerToken = new JwtSecurityToken(token.EncodedToken);
+                outerToken.InnerToken = new JwtSecurityToken(token.InnerToken.EncodedToken);
+                return outerToken;
+            }
+            else if (!string.IsNullOrEmpty(token.EncodedToken))
+            {
+                return new JwtSecurityToken(token.EncodedToken);
+            }
+            else if (token.Header != null && token.Payload != null)
```

Comment:
If we fix the bug where JsonWebToken.EncodedToken is never null, we will not need this.
Also, you could write: new JwtSecurityToken(token.EncodedHeader + "." token.EncodedPayload + ".")

Reply:
Believe this is addressed, @brentschmaltz what do you think?